src/core/ext/lb_policy/round_robin/round_robin.c

```diff
@@ -416,6 +415,7 @@ static int rr_pick(grpc_exec_ctx *exec_ctx, grpc_lb_policy *pol,
     }
     /* only advance the last picked pointer if the selection was used */
     advance_last_picked_locked(p);
+    gpr_mu_unlock(&p->mu);
```

Comment:
Why is this change needed?  I'm probably missing something obvious, but I'd like to make sure I understand the reason for the change.